contracts/interchain-token-service/src/contract/execute/mod.rs

```diff
@@ -100,12 +111,54 @@ pub fn execute_message(
     }
 }
 
+// messages sent to axelar are handled in a special way
+// right now, messages sent to the axelar chain are not forwarded on to
+// any other contract (in contrast to every other message that moves through the hub)
+// In the future, this may change, depending on the message type
+// The main use case for this at the moment is the RegisterToken message,
+// which simply informs the ITS hub of the decimals and token address of a
+// custom token, and thus needs no forwarding.
+fn execute_axelar_message(
+    deps: DepsMut,
+    cc_id: CrossChainId,
+    destination_chain: ChainNameRaw,
+    message: Message,
+) -> Result<Response, Error> {
+    let message = apply_to_hub(
+        deps.storage,
+        cc_id.source_chain.clone(),
+        destination_chain.clone(),
+        message,
+    )?;
+    Ok(Response::new().add_event(
+        Event::MessageReceived {
+            cc_id,
+            destination_chain,
+            message,
+        }
+        .into(),
+    ))
+}
```

Comment:
idea to unify the behaviour and get rid of this special case:
- move loading the destination address and destination payload into the `send_to_destination` function
- add the special case to send_to_destination, that if the destination is axelarnet then be a noop
as far as I can tell, then execute_message_on_hub does the same as execute_axelar_message

Reply:
we would also need to somehow skip the frozen check, since axelar is not a registered chain. Or somehow register axelar as a chain, even though there is no ITS contract yet on Axelar